client/rest-high-level/src/test/java/org/elasticsearch/client/CrudIT.java

```diff
@@ -265,4 +340,27 @@ public void testIndex() throws IOException {
                          "version conflict, document already exists (current version [1])]", exception.getMessage());
         }
     }
+
+    /**
+     * @return A sample index request
+     * @throws IOException If problem when generating the content
+     */
+    private IndexRequest createSampleIndexRequest() throws IOException {
+        final XContentType xContentType = randomFrom(XContentType.values());
+        IndexRequest indexRequest = new IndexRequest("index", "type");
+        indexRequest.source(XContentBuilder.builder(xContentType.xContent()).startObject().field("test", "test").endObject());
```

Comment:
if this is only used for testing deletions, I don't think we need to randomize the content type when indexing. Let's use json? we can also provide it as a string.